api/v1beta1/machine_types.go

```diff
@@ -230,10 +230,32 @@ type MachineStatus struct {
 	// Conditions defines current service state of the Machine.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// Deletion contains information relating to removal of the Machine.
+	// Only present when the Machine has a deletionTimestamp and is being removed from the cluster.
```

Comment:
Not sure what "is being removed from the cluster" means in this context

Is this something in addition to the deletionTimestamp?

Reply:
I guess we need to clarify when a deletionTimestamp doesn't mean being removed, ie hooks present/blocking us getting to the drain phase?

Reply:
Updating it to:
```
// Only present when the Machine has a deletionTimestamp and and drain or wait for volume detach started.
```

Because there can be cases where e.g. drain is skipped entirely :-)